### ISSUE TITLE
Panic with malformed _geo data

### ISSUE THREAD

**Issue by laurentS** [meilisearch/meilisearch#4333]
**Describe the bug**
In v1.6, indexing malformed `_geo` data causes the server to panic and crash (without the [error](https://www.meilisearch.com/docs/reference/errors/error_codes#invalid_document_geo_field) listed in the [docs](https://www.meilisearch.com/docs/learn/fine_tuning_results/geosearch#preparing-documents-for-location-based-search)).

**To Reproduce**
1. `docker run -it --rm -p 7700:7700 -e MEILI_ENV='development' -e RUST_BACKTRACE=full -v $(pwd)/meili_data:/meili_data getmeili/meilisearch:v1.6`
2. Upload the following json document:
```json
{"id":25843,"url_name":"mairie_champmillon","name":"Mairie - Champmillon","rip":false,"category":"mairie","addresse":"Route de la Charente, 16290, Champmillon","cp":16290,"_geo":{"lng":null,"lat":null}}
```
4. (I use the python client) `client.index("myindex").update_filterable_attributes(["_geo"])
`
5. See error:
```
[2024-01-16T16:37:17Z INFO  actix_web::middleware::logger] 172.17.0.1 "PUT /indexes/myindex/settings/filterable-attributes HTTP/1.1" 202 126 "-" "Meilisearch Python (v0.28.4)" 0.025347
thread 'indexing-thread:7' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected value", line: 1, column: 1)', milli/src/update/index_documents/extract/extract_geo_points.rs:38:49
stack backtrace:
   0:     0x562c77d4a411 - std::backtrace_rs::backtrace::libunwind::trace::h2801c3d891641b62
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x562c77d4a411 - std::backtrace_rs::backtrace::trace_unsynchronized::h9d8e6ca1d0a50926
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x562c77d4a411 - std::sys_common::backtrace::_print_fmt::h4ca85c9e04b799d6
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x562c77d4a411 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h25eb23ac136a62ea
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x562c77d75bff - core::fmt::rt::Argument::fmt::h4eea7b4e1cd3d54c
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/fmt/rt.rs:138:9
   5:     0x562c77d75bff - core::fmt::write::h8ff10c105165a0ca
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/fmt/mod.rs:1094:21
   6:     0x562c77d45fd1 - std::io::Write::write_fmt::h0b273f3dbe587d74
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/io/mod.rs:1713:15
   7:     0x562c77d4a225 - std::sys_common::backtrace::_print::h9ce65571a3fb9232
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x562c77d4a225 - std::sys_common::backtrace::print::h6daade9b653ced18
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x562c77d4b6f7 - std::panicking::default_hook::{{closure}}::h434e1f618d54ba39
  10:     0x562c77d4b4e4 - std::panicking::default_hook::ha7b0fd99da48bd33
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/panicking.rs:288:9
  11:     0x562c77d4bbac - std::panicking::rust_panic_with_hook::hf213409f84e89d28
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/panicking.rs:705:13
  12:     0x562c77d4baa7 - std::panicking::begin_panic_handler::{{closure}}::h8bb9686ec9223485
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/panicking.rs:597:13
  13:     0x562c77d4a846 - std::sys_common::backtrace::__rust_end_short_backtrace::h1b1a98c6c5aa5371
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/sys_common/backtrace.rs:151:18
  14:     0x562c77d4b7f2 - rust_begin_unwind
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/panicking.rs:593:5
  15:     0x562c76a28183 - core::panicking::panic_fmt::hd33dcf850a5fb2b4
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/panicking.rs:67:14
  16:     0x562c76a28623 - core::result::unwrap_failed::h9fffefee0d2db1ed
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/result.rs:1651:5
  17:     0x562c77522d46 - milli::update::index_documents::extract::extract_geo_points::extract_geo_points::{{closure}}::h18519b2f114f3138
  18:     0x562c774aefab - milli::update::index_documents::extract::extract_geo_points::extract_lat_lng::h9b0aee5131efb6a3
  19:     0x562c773b526a - <rayon_core::job::HeapJob<BODY> as rayon_core::job::Job>::execute::h5d049ab5d9036d42
  20:     0x562c76a11324 - rayon_core::registry::WorkerThread::wait_until_cold::h747b490e5ee66189
  21:     0x562c77a2f5e2 - rayon_core::registry::ThreadBuilder::run::hde8c8da50f1df35c
  22:     0x562c77a2d0da - std::sys_common::backtrace::__rust_begin_short_backtrace::h6c13c1509fd916ae
  23:     0x562c77a2d603 - core::ops::function::FnOnce::call_once{{vtable.shim}}::hd9f92199f0ff7236
  24:     0x562c77d4ff65 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hb71d4e640321734e
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/alloc/src/boxed.rs:1985:9
  25:     0x562c77d4ff65 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h27df4586128624e9
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/alloc/src/boxed.rs:1985:9
  26:     0x562c77d4ff65 - std::sys::unix::thread::Thread::new::thread_start::ha304f449462101a7
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/sys/unix/thread.rs:108:17
  27:     0x7fd5616601d9 - <unknown>
Rayon: detected unexpected panic; aborting
```

**Expected behavior**
The data is invalid, but the server shouldn't panic this way, a useful error message would be much more helpful.

**Screenshots**

**Meilisearch version:**
v.1.6.0 (docker). The same data was uploaded and apparently indexed successfully with v1.5.1.

**Additional context**
Using python client, though I don't think this matters.

**Comment by curquiza** [comment]
Hello @laurentS 
Thanks for the report, our team is working on fixing it!

**Comment by irevoire** [comment]
Hey, thanks for the report @laurentS.
It looks like we had a lot of tests on the addition of geo documents but not much on the errors while updating the settings for the geo-faceted documents.

I added a few more tests and fixed your issue we should be good.

Also @curquiza, this bug was introduced with the diff indexing in the last release

**Comment by laurentS** [comment]
Wow, thanks for the super fast fix!

**Comment by irevoire** [comment]
Actually, the bug is worse than I thought thus, we're going to fix it for the v1.6.1 in a patch release

**Comment by curquiza** [comment]
fixed by #4366
